ambry-api/src/main/java/com.github.ambry/account/Container.java

```diff
@@ -210,38 +272,87 @@ private Container(JSONObject metadata) throws JSONException {
+        parentAccountId = (short) metadata.getInt(PARENT_ACCOUNT_ID_KEY);
+        encrypted = metadata.optBoolean(ENCRYPTED_KEY, ENCRYPTED_DEFAULT_VALUE);
+        previouslyEncrypted = metadata.optBoolean(PREVIOUSLY_ENCRYPTED_KEY, PREVIOUSLY_ENCRYPTED_DEFAULT_VALUE);
+        cacheable = metadata.optBoolean(CACHEABLE_KEY, CACHEABLE_DEFAULT_VALUE);
+        mediaScanDisabled = metadata.optBoolean(MEDIA_SCAN_DISABLED, MEDIA_SCAN_DISABLED_DEFAULT_VALUE);
         break;
-
       default:
         throw new IllegalStateException("Unsupported container json version=" + metadataVersion);
     }
+    checkPreconditions();
   }
 
   /**
    * Constructor that takes individual arguments. Cannot be null.
-   * @param id The id of the container. Cannot be null.
+   * @param id The id of the container.
    * @param name The name of the container. Cannot be null.
    * @param status The status of the container. Cannot be null.
    * @param description The description of the container. Can be null.
-   * @param isPrivate The privacy setting of the container. Cannot be null.
-   * @param parentAccountId The id of the parent {@link Account} of this container. Cannot be null.
-   */
-  Container(Short id, String name, ContainerStatus status, String description, Boolean isPrivate,
-      Short parentAccountId) {
+   * @param encrypted {@code true} if blobs in the {@link Container} should be encrypted, {@code false} otherwise.
+   * @param previouslyEncrypted {@code true} if this {@link Container} was encrypted in the past, or currently, and a
+   *                            subset of blobs in it could still be encrypted.
+   * @param cacheable {@code true} if cache control headers should be set to allow CDNs and browsers to cache blobs in
+   *                  this container.
+   * @param mediaScanDisabled {@code true} if media scanning for content in this container should be disabled.
+   * @param parentAccountId The id of the parent {@link Account} of this container.
+   */
+  Container(short id, String name, ContainerStatus status, String description, boolean encrypted,
+      boolean previouslyEncrypted, boolean cacheable, boolean mediaScanDisabled, short parentAccountId) {
     this.id = id;
     this.name = name;
     this.status = status;
     this.description = description;
-    this.isPrivate = isPrivate;
+    this.encrypted = encrypted;
+    this.previouslyEncrypted = previouslyEncrypted;
+    this.cacheable = cacheable;
+    this.mediaScanDisabled = mediaScanDisabled;
     this.parentAccountId = parentAccountId;
-    checkRequiredFields();
+    checkPreconditions();
+  }
+
+  /**
+   * Deserializes a {@link JSONObject} to a container object.
+   * @param json The {@link JSONObject} to deserialize.
+   * @return A container object deserialized from the {@link JSONObject}.
+   * @throws JSONException If parsing the {@link JSONObject} fails.
+   */
+  public static Container fromJson(JSONObject json) throws JSONException {
```

Comment:
does this need to be `public`?

Reply:
changed it back to package private. Wanted to avoid requiring people to use the builder when it wasn't needed (when all fields were supplied), but that will create a lot of changes so i will do that in a future pr.

Reply:
This can be package private, since it is only used by Account